package/MDAnalysis/core/Selection.py

```diff
@@ -543,8 +518,7 @@ class ProteinSelection(Selection):
 
     .. SeeAlso:: :func:`MDAnalysis.lib.util.convert_aa_code`
     """
-    #: Dictionary of recognized residue names (3- or 4-letter).
-    prot_res = {x: None for x in [
+    prot_res = [
```

Comment:
And it is *a lot* (factor 5)  faster:
```
In [1]: names_list = [# CHARMM top_all27_prot_lipid.rtf
   ...:         'ALA', 'ARG', 'ASN', 'ASP', 'CYS', 'GLN', 'GLU', 'GLY', 'HSD',
   ...:         'HSE', 'HSP', 'ILE', 'LEU', 'LYS', 'MET', 'PHE', 'PRO', 'SER', 'THR',
   ...:         'TRP', 'TYR', 'VAL', 'ALAD',
   ...:         ## 'CHO','EAM', # -- special formyl and ethanolamine termini of gramicidin
   ...:         # PDB
   ...:         'HIS', 'MSE',
   ...:         # from Gromacs 4.5.3 oplsaa.ff/aminoacids.rtp
   ...:         'ARGN', 'ASPH', 'CYS2', 'CYSH', 'QLN', 'PGLU', 'GLUH', 'HIS1', 'HISD',
   ...:         'HISE', 'HISH', 'LYSH',
   ...:         # from Gromacs 4.5.3 gromos53a6.ff/aminoacids.rtp
   ...:         'ASN1', 'CYS1', 'HISA', 'HISB', 'HIS2',
   ...:         # from Gromacs 4.5.3 amber03.ff/aminoacids.rtp
   ...:         # Amber: there are also the C-term aas: 'CALA', 'CGLY', 'CSER', ...
   ...:         # Amber: there are also the N-term aas: 'NALA', 'NGLY', 'NSER', ...
   ...:         'HID', 'HIE', 'HIP', 'ORN', 'DAB', 'LYN', 'HYP', 'CYM', 'CYX', 'ASH',
   ...:         'GLH',
   ...:         'ACE', 'NME',]

In [2]: names_dict = {x: None for x in names_list}
In [4]: %timeit 'TYR' in names_list
1000000 loops, best of 3: 560 ns per loop

In [5]: %timeit 'TYR' in names_dict
10000000 loops, best of 3: 96.2 ns per loop
```

The spee-up probably does not matter here but on grounds of principle I would leave it a dict.